### ISSUE TITLE
Mustang doesn't compile with latest nightly

### ISSUE THREAD

**Issue by sunfishcode** [sunfishcode/mustang#138]
Compiling mustang with the latest nightly gets these errors:
```
inlinable function call in a function with debug info must have a !dbg location
  call void @_Unwind_Resume(ptr %81) #15
inlinable function call in a function with debug info must have a !dbg location
  call void @_Unwind_Resume(ptr %101) #15
inlinable function call in a function with debug info must have a !dbg location
  call void @_Unwind_Resume(ptr %144) #15
inlinable function call in a function with debug info must have a !dbg location
  call void @_Unwind_Resume(ptr %111) #15
LLVM ERROR: Broken module found, compilation aborted!
```

**Comment by sunfishcode** [comment]
Compiling with `--release` works.
Compiling with the 2022-08-07 snapshot works.

**Comment by sunfishcode** [comment]
This might be due to https://github.com/rust-lang/rust/issues/100783.

**Comment by Urgau** [comment]
This is this issue: https://github.com/rust-lang/rust/issues/101121

**Comment by Urgau** [comment]
Normally fixed in the 3 latest nightlies (starting from 4 September)

**Comment by Urgau** [comment]
Running `cargo +nightly run -Z build-std --target=mustang/target-specs/x86_64-mustang-linux-gnu.json --example hello` works for me on the two latest nightlies (`rustc 1.65.0-nightly (78a891d36 2022-09-05)` and `rustc 1.65.0-nightly (78a891d36 2022-09-06)`).

**Comment by sunfishcode** [comment]
Ah, thanks! My rustc wasn't updated because rustup was holding it back because there isn't a new miri build. #137 is now using the latest nightly.

**Comment by sunfishcode** [comment]
This appears fixed in #140.
